### PR TITLE
Introduces the "match" extern function for strings

### PR DESCRIPTION
**Pull request by ozevren**
**Release note**:

```release-note
NONE
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1240)

### CODE REVIEW

**Comment by ozevren** [comment]
Codecov numbers might be deficient (I can't run it locally). I'll beef up tests if necessary if the numbers come out worse.

**Comment by ozevren** [comment]
There used to be codecov report. Not anymore, I guess?

File: pkg/expr/expr_test.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ func TestGoodParse(t *testing.T) {
 		{`a/b`, `QUO($a, $b)`},
 		{`request.header["X-FORWARDED-HOST"] == "aaa"`, `EQ(INDEX($request.header, "X-FORWARDED-HOST"), "aaa")`},
 		{`source.ip | ip("0.0.0.0")`, `OR($source.ip, ip("0.0.0.0"))`},
+		{`matches(service.name, "cluster1.ns.*")`, `matches($service.name, "cluster1.ns.*")`},
```
Comment (mandarjog) [inline pkg/expr/expr_test.go]: can you rename the function `match` ?

Comment (ozevren) [reply pkg/expr/expr_test.go]: Done

**Comment by mandarjog** [comment]
You should be able to run coverage tests locally, can you just post your results? 

@yutongz for some reason codecov is not posting.

**Comment by ozevren** [comment]
I can't run them locally. There is always some compilation error or other.

**Comment by yutongz** [comment]
I was also confused. From what I saw in [codecov history](https://codecov.io/gh/istio/mixer/commits?&page=6), it seems from [#1172](https://codecov.io/gh/istio/mixer/commit/1da71b6ce98f1708496327952dd7e851e5525899), all commits failed to run the script to calculate code coverage.
From [CI log](https://storage.googleapis.com/istio-prow/pull/istio_mixer/1172/mixer-presubmit/565/build-log.txt), it complained the import name issue.
```
W0901 02:29:33.303] + echo '=== Code Coverage ==='
W0901 02:29:33.303] + ./bin/codecov.sh
W0901 02:29:33.303] + tee codecov.report
W0901 02:29:33.722] can't load package: adapter/inventory.gen.go:22:2: code in directory /home/bootstrap/go/src/istio.io/mixer/adapter/memquota2 expects import "istio.io/mixer/adapter/memquota"
W0901 02:29:33.722] can't load package: package istio.io/mixer/adapter/statsd2: code in directory /home/bootstrap/go/src/istio.io/mixer/adapter/statsd2 expects import "istio.io/mixer/adapter/statsd"
W0901 02:29:34.132] adapter/inventory.gen.go:22:2: code in directory /home/bootstrap/go/src/istio.io/mixer/adapter/memquota2 expects import "istio.io/mixer/adapter/memquota"
```
Do you guys have any idea about this?

**Comment by mandarjog** [comment]
Aah, ok , I will investigate @yutongz

**Pull request merged**